base/pmap.jl

```diff
@@ -158,15 +170,24 @@ function wrap_on_error(f, on_error; capture_data=false)
     end
 end
 
-wrap_retry(f, retry_delays, retry_check) = retry(f, delays=retry_delays, check=retry_check)
+function wrap_retry(f, retry_delays, retry_check)
+    f2 = x -> begin
+        try
+            f(x)
+        catch e
+            rethrow(extract_exception(e))
+        end
+    end
+    retry(f2, delays=retry_delays, check=retry_check)
```

Comment:
Not a big deal, but this could perhaps be written more succinctly using a `do` block, e.g.
```julia
retry(delays=retry_delays, check=retry_check) do x
    try
        f(x)
    catch e
        rethrow(extract_exception(e))
    end
end
```